File: docs-examples/example-java/src/test/java/io/micronaut/rabbitmq/docs/rpc/ProductListener.java
package io.micronaut.rabbitmq.docs.rpc;

import io.micronaut.context.annotation.Requires;
// tag::imports[]
import io.micronaut.rabbitmq.annotation.Queue;
import io.micronaut.rabbitmq.annotation.RabbitListener;
// end::imports[]

@Requires(property = "spec.name", value = "RpcUppercaseSpec")
// tag::clazz[]
@RabbitListener
public class ProductListener {

    @Queue("product")
    public String toUpperCase(String data) { // <1>
        return data.toUpperCase(); // <2>
    }
}
// end::clazz[]

File: docs-examples/example-kotlin/src/test/kotlin/io/micronaut/configuration/rabbitmq/docs/rpc/ProductListener.kt
package io.micronaut.configuration.rabbitmq.docs.rpc

// tag::imports[]
import io.micronaut.configuration.rabbitmq.annotation.Queue
import io.micronaut.configuration.rabbitmq.annotation.RabbitListener
import io.micronaut.context.annotation.Requires

import java.io.IOException
// end::imports[]

@Requires(property = "spec.name", value = "RpcUppercaseSpec")
// tag::clazz[]
@RabbitListener
class ProductListener {

    @Queue("product")
    @Throws(IOException::class)
    fun toUpperCase(data: String): String { // <1>
        return data.toUpperCase() // <2>
    }
}
// end::clazz[]
